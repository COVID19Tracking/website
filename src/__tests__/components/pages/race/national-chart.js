import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import NationalChart from '~components/pages/race/national-chart'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    covidRaceDataHomepage: {
      whiteMortalityRate: '25.4',
      twoMortalityRate: '2.5',
      otherMortalityRate: '25.1',
      nhpiMortalityRate: '20.9',
      latinXMortalityRate: '29.7',
      blackMortalityRate: '64.7',
      aianMortalityRate: '30.6',
      blackLivesLost: '26708',
      blackPercentOfDeath: '0.2329585598',
    },
    covidRaceDataTimeseries: {
      Date: '20200101'
    }
  }))
})

describe('Components : Race : National Chart', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NationalChart />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
