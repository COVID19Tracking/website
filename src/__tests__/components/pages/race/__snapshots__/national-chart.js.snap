// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Race : National Chart renders correctly 1`] = `
<div>
  <div
    className="charts"
  >
    <div
      className="header"
    >
      <div
        className="spacer"
      />
      <span>
        Deaths per 
        100,000
         people by race or ethnicity
      </span>
    </div>
    <div
      className="raceRow emphasis"
    >
      <div>
        <span
          className="rowLabel"
        >
          Black or African American
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "75%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        Black or African American
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        65
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow"
    >
      <div>
        <span
          className="rowLabel"
        >
          American Indian or Alaskan Native
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "35.47140649149923%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        American Indian or Alaskan Native
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        31
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow"
    >
      <div>
        <span
          className="rowLabel"
        >
          Hispanic or Latino
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "34.42812982998454%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        Hispanic or Latino
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        30
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow emphasis"
    >
      <div>
        <span
          className="rowLabel"
        >
          White
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "29.4435857805255%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        White
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        25
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow"
    >
      <div>
        <span
          className="rowLabel"
        >
          Other
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "29.09582689335394%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        Other
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        25
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow"
    >
      <div>
        <span
          className="rowLabel"
        >
          Native Hawaiian or Other Pacific Islander
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "24.227202472952087%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        Native Hawaiian or Other Pacific Islander
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        21
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
    <div
      className="raceRow"
    >
      <div>
        <span
          className="rowLabel"
        >
          Two or more races
        </span>
      </div>
      <div
        className="bar"
        style={
          Object {
            "flexBasis": "2.8979907264296756%",
          }
        }
      />
      <span
        className="a11y-only"
      >
        Two or more races
         people have experienced
      </span>
      <span
        className="mortalityRateLabel"
      >
        3
      </span>
      <span
        className="a11y-only"
      >
        deaths per 
        100,000
        .
      </span>
    </div>
  </div>
  <div
    className="disclosureWrapper"
  >
    <div
      className="spacer"
    />
    <div
      className="disclosureContainer"
    >
      <button
        aria-controls="panel--1"
        aria-expanded={false}
        className="showNotes"
        data-reach-disclosure-button=""
        data-state="collapsed"
        onClick={[Function]}
      >
        Notes 
        <span
          aria-hidden={true}
        >
          ↓
        </span>
      </button>
      <div
        className="notes"
        data-reach-disclosure-panel=""
        data-state="collapsed"
        hidden={true}
        id="panel--1"
        tabIndex={-1}
      >
        These calculations are based on data from The Covid Racial Data Tracker and the U.S. Census Bureau. Race categories may overlap with Hispanic/Latinx ethnicity. Rates are not age-adjusted and some rates are underestimated due to lack of reporting of race and ethnicity categories for COVID-19 deaths.
      </div>
    </div>
  </div>
  <p
    className="paragraph paragraph"
  >
    We’ve lost at least
     
    26708
     Black lives to COVID-19 to date. Black people account for
     
    23
    % of COVID-19 deaths where race is known.
  </p>
</div>
`;
